tools/replace_code_block.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from traceback import print_tb
+from typing import AnyStr
+
+
+def check_for_code(filename: AnyStr, check: bool, print_not_fixed: bool):
+    """Check a file and replace ``.. code::`` to ``.. code-block::``
+
+    :param filename: The file to open
+    :param check: Whether or not the program is operating in ``check`` mode.
+        In check mode, the program will exit with a status 1 if files will be
+        replaced. In non-check mode, the program will replace files.
+    :param print_not_fixed: Print files that were not re-formatted, similar to
+        black.
+    """
+    status = 0
+    with open(filename, "rb") as fp:
```

Comment:
As suggested, let's not open these files in binary mode. If we have `*.py` files which are encoded in a non-standard format, I consider that to be an issue we want to be brought up.

Reply:
Understood. I will replace this to open utf-8 encoded files